apluslms_roman/builder.py

```diff
@@ -36,22 +36,24 @@ def build(self, step_refs: list = None, clean_build=False):
         backend = self._engine.backend
         observer = self._observer
         steps = self.get_steps(step_refs) # NOTE: may raise KeyError or IndexError
-
-        task = BuildTask(self.path, steps)
-        observer.enter_prepare()
-        result = backend.prepare(task, observer)
-        observer.result_msg(result)
-        if result.ok:
-            observer.enter_build()
-            # FIXME: add support for other build paths
-            if clean_build:
-                if isdir('_build'):
-                    rmtree('_build')
-            if clean_build or not isdir('_build'):
-                mkdir('_build')
-            result = backend.build(task, observer)
+        try:
+            task = BuildTask(self.path, steps)
+            observer.enter_prepare()
+            result = backend.prepare(task, observer)
             observer.result_msg(result)
-        observer.done(result)
+            if result.ok:
+                observer.enter_build()
+                # FIXME: add support for other build paths
+                if clean_build:
+                    if isdir('_build'):
+                        rmtree('_build')
+                if clean_build or not isdir('_build'):
+                    mkdir('_build')
+                result = backend.build(task, observer)
+                observer.result_msg(result)
+            observer.done(result)
+        except KeyError:
```

Comment:
Is this supposed to be KeyboardInterrupt?

Reply:
...wonder what happened there. Probably just started writing with Key and then used autocomplete and didn't think much about it :')